File: HackerRank/Array/array_manipulation.cpp
#include<iostream>
using namespace std;
int main()
{
    int size, no;
    cin>>size>>no;
    int arr[size];
    for(int i=0;i<size;i++)
        arr[i]=0;
    for(int i=0;i<no;i++)
    {
        int m,n,op;
        cin>>m>>n>>op;
        m--;
        n--;

        for(int j=m;j<=n;j++)
        {
            arr[j]+=op;
        }
    }

}

File: HackerRank/Array/array_manipulation.py
n,m=map(int, input().split(' '))
arr=[0.0]*n
for _ in range(m):
    a, b, k = map(int, input().split(' '))
    a=a-1
    b=b-1
    for i in range(a,b+1):
        arr[i]=arr[i]+k


maxi=float('-inf')
for i in range(n):
    if arr[i]>=maxi:
        maxi=arr[i]
print(int(maxi))